plugins/uefi/fu-uefi-common.c

```diff
@@ -241,12 +244,17 @@ fu_uefi_get_esp_path_for_os (const gchar *base)
 		return g_steal_pointer (&esp_path);
 	/* if ID key doesn't exist, try ID_LIKE */
 	id_like_id = g_hash_table_lookup (os_release, "ID_LIKE");
+	/* try to fallback to use UEFI removable path if ID_LIKE key doesn't exist */
 	if (id_like_id != NULL) {
 		g_autofree gchar* id_like_path = g_build_filename (base, "EFI", id_like_id, NULL);
 		if (g_file_test (id_like_path, G_FILE_TEST_IS_DIR)) {
 			g_debug ("Using ID_LIKE key from os-release");
 			return g_steal_pointer (&id_like_path);
 		}
+	} else if (fu_device_get_metadata_boolean (device, "FallbacktoRemovablePath")) {
+		esp_path = g_build_filename (base, "EFI", "boot", NULL);
+		if (g_file_test (esp_path, G_FILE_TEST_IS_DIR))
+			return g_steal_pointer (&esp_path);
```

Comment:
seems to me that this is not necessary, the next line (259) does it unconditionally.

Reply:
How about this?